lib/mercury/fake.rb

```diff
@@ -43,7 +45,15 @@ def close(&k)
 
     def publish(source_name, msg, tag: '', headers: {}, &k)
       guard_public(k)
-      queues.values.select{|q| q.binds?(source_name, tag)}.each{|q| q.enqueue(roundtrip(msg), tag, headers)}
+      queues.values.select{|q| q.binds?(source_name, tag)}.each{|q| q.enqueue(roundtrip(msg), tag, headers.stringify_keys)}
+      ret(k)
+    end
+
+    def republish(msg, &k)
+      guard_public(k)
+      msg.ack
```

Comment:
Yeah, remember that it's just a simulation. How it actually works doesn't matter, just what is observed.